### DISCUSSION TITLE
Nordigen API for transaction data

### DISCUSSION THREAD

**Discussion by dabeeeenster** [firefly-iii/firefly-iii#4479] [General]
Just found this: https://nordigen.com/en/

Could this be used as a free solution for account transaction data within Europe?

**Comment by JC5** [comment]
Looks like. I've signed up, see what they'll deliver.

**Comment by dabeeeenster** [comment]
Did this get anywhere?

**Reply by JC5** [reply]
I have an account now, and a bank i'm a member of is onboarded with Nordigen. So I can check it out and see if it works. I won't build stuff for Firefly III though, I have other priorities at the moment.

**Reply by dabeeeenster** [reply]
👍

**Reply by JC5** [reply]
Then again, I'll put it on GitHub anyway, i know myself. But no fancy importers for the time being.

**Reply by diogotcorreia** [reply]
I am building an importer, not sure how long it will take tho. Feel free to contribute with ideas or code: https://github.com/diogotcorreia/nordigen2fireflyiii

PS: I'm using this as a way to learn Rust, so it's likely gonna take a while since I don't know the language

**Reply by dabeeeenster** [reply]
This looks like a great start! How were you planning on interfacing with Firefly? Using the same API as the existing docker uploader?

**Reply by diogotcorreia** [reply]
@dabeeeenster Yes! When I was researching I found the API docs for Firefly. I'm guessing it's the same that's used in the CSV Importer.
For now, I'm still getting the data off Nordigen, I haven't really thought on how I'm going to handle data going into Firefly

**Reply by dabeeeenster** [reply]
Initially you could just write it to CSV the use the existing importer?

**Reply by JC5** [reply]
Yes, the API is used by all importers, no custom stuff!

**Reply by diogotcorreia** [reply]
I might end up adding an option to export to CSV first, seems a bit easier than doing a full firefly import!
Thanks for the suggestion

**Reply by ghost** [reply]
Hi! Any updates on this?

**Reply by JC5** [reply]
I'm working on an importer.

**Reply by axelsimon** [reply]
Which now works. Let's mark this as answered!
